src/modules/AutoinstClass.rb

```diff
@@ -276,10 +234,10 @@ def compat_class_file
     # Builds a map of classes to import from /etc/autoinstall/classes.xml
     # @return [Array<Hash>] Classes defined in the file.
     def read_old_classes
-      old_classes_map = Convert.to_map(SCR.Read(path('.xml'), compat_class_file))
-      old_classes = old_classes_map['classes'] || []
+      old_classes_map = Convert.to_map(SCR.Read(path(".xml"), compat_class_file))
+      old_classes = old_classes_map["classes"] || []
```

Comment:
are you sure that previous call cannot return nil? I think if file do nto exists it can be problem and also if xml contain different stuff then map.